### ISSUE TITLE
CI/CD: flaky tests

### ISSUE THREAD

**Issue by yashrsharma44** [thanos-io/thanos#3641]
While running GitHub Action tests for any Pull Request changes, the receiver tests somehow fail irrespective of the codechange made(or documentation changes). Seems like an example of a flaky test - 
For Example - Flaky test observed in this #3630 

```powershell
FAIL
FAIL	github.com/thanos-io/thanos/test/e2e	111.932s
?   	github.com/thanos-io/thanos/test/e2e/e2ethanos	[no test files]
FAIL
make: *** [test-e2e] Error 1
Error: Makefile:234: recipe for target 'test-e2e' failed
Error: Process completed with exit code 2.
```

**Comment by bwplotka** [comment]
This is to broad. Let's focus on particular test which is flaky.

**Comment by prmsrswt** [comment]
I agree with Bartek. The specific case you are pointing to already has issue #2622 that is already closed (by #3798), so I am closing this issue for now as well as I think we don't have anything actionable here now. Let's reopen it if that's not the case.
